src/modules/stripe/stripe.service.ts

```diff
@@ -0,0 +1,134 @@
+  private stripe: Stripe;
+  private readonly baseUrl = this.configService.get('ALLOWED_ORIGINS');
+
+  constructor(
+    private readonly configService: ConfigService,
+    private readonly bookingsService: BookingsService,
+    private readonly accommodationsService: AccommodationsService,
+  ) {
+    const apiKey = this.configService.get('STRIPE_SECRET_KEY');
+    this.stripe = new Stripe(apiKey);
+  }
+
+  async checkout(dto: CreateCheckoutDto) {
+    const accommodation = await this.accommodationsService.getAccommodationById(
+      dto.accommodationId,
+    );
+    const booking = await this.bookingsService.getBookingById(dto.bookingId);
+
+    return await this.createCheckoutSession(accommodation, booking);
+  }
+
+  private async createCheckoutSession(
+    accommodation: Accommodation,
+    booking: Booking,
+  ) {
+    const accommodationImage = accommodation.images
+      .filter((image) => image.order === 1)
+      .map((image) => image.url);
+    const session = await this.stripe.checkout.sessions.create({
+      line_items: [
+        {
+          price_data: {
+            currency: 'usd',
+            product_data: {
+              name: accommodation.name,
+              description: accommodation.description,
+              images: accommodationImage,
+            },
+            unit_amount: booking.totalPrice * 100,
+          },
+          quantity: 1,
+        },
+      ],
+      mode: 'payment',
+      metadata: {
+        bookingId: booking.id,
+        userId: booking.user.id,
+      },
+      // For now we redirect to the same page with query params
+      success_url: `${this.baseUrl}/property/${accommodation.id}?success=true&bookingId=${booking.id}`,
+      cancel_url: `${this.baseUrl}/property/${accommodation.id}?success=false&bookingId=${booking.id}`,
+    });
+
+    return { checkoutSessionUrl: session.url };
+  }
+
+  async handleWebhook(body: Buffer, signature: string) {
+    const event = this.constructEvent(body, signature);
+    if (!event) return { received: false };
+
+    switch (event.type) {
+      case 'payment_intent.succeeded':
+        const paymentIntentSuccess = event.data.object;
+        return await this.handlePaymentIntent(paymentIntentSuccess, 'success');
+      case 'payment_intent.payment_failed':
+        const paymentIntentFailed = event.data.object;
+        return await this.handlePaymentIntent(paymentIntentFailed, 'failed');
+      default:
+        throw new BadRequestException(`Unhandled event type ${event.type}`);
+    }
+  }
+
+  async handlePaymentIntent(
+    paymentIntent: Stripe.PaymentIntent,
+    paymentStatus: 'success' | 'failed',
+  ) {
+    const metadata = paymentIntent.metadata;
+    const bookingId = metadata.bookingId;
+    const userId = metadata.userId;
+
+    const booking = await this.bookingsService.getBooking(userId, bookingId);
+
+    if (!booking) {
+      throw new BadRequestException(`Booking with id ${bookingId} not found`);
+    }
```

Comment:
Maybe booking exists but it doesn't belong to the user? It would be better to check that case as well and send a proper message.